src/features/unified-share-modal/ContactsField.js

```diff
@@ -127,6 +127,19 @@ class ContactsField extends React.Component<Props, State> {
 
     debouncedGetContacts = debounce(this.getContactsPromise, 200);
 
+    handleParseItems = (inputValue: string): Array<string> => {
+        const { validator } = this.props;
+
+        // ContactField allows invalid pills to be displayed in
+        // in some cases (e.g., when user is external and external
+        // collab is restricted). We don't allow, however, invalid
+        // emails from the pill selector input to be turned into pills.
+        const emails = parseEmails(inputValue);
```

Comment:
Previously, invalid emails on the pill selector input were filtered out by `PillSelectorDropdown` automatically, but that filtering step is skipped when `allowInvalidPills` is enabled:

https://github.com/box/box-ui-elements/blob/master/src/components/pill-selector-dropdown/PillSelectorDropdown.js#L117

By filtering out invalid emails on the pill parser, we can keep the existing behavior of preventing invalid emails from generating pills, but also allow invalid pills to be shown for other scenarios.